sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -348,7 +356,21 @@ struct graphs_support;
 } // namespace device
 } // namespace info
 
-class node {};
+
+class node {
+  public:
+  class dynamic_parameter{
+    dynamic_parameter() = delete;
+    const std::string& getLabel() const;
+  };
+
+  node() = delete;
+
+	node_type getType() const;
+  std::vector<Node> getDependencies() const;
```

Comment:
Strictly no it's not required, however without it the use of this would potentially be limited to only applications which purely use the explicit API, or the limited mixed usage we already support. Queries like this I think are very useful for helping the user reason about nodes in a graph obtained via `getNodes()` which allows them to use nodes recorded via queue submission fully with the explicit API.

This doesn't cover all uses or information required probably but is at least a step in that direction.

Reply:
I've added a `getDependentNodes` query, good idea!